helix-lsp/src/lib.rs

```diff
@@ -204,6 +204,19 @@ pub mod util {
         // in reverse order.
         edits.sort_unstable_by_key(|edit| edit.range.start);
 
+        if edits.len() == 1 {
```

Comment:
Is it possible for a text edit that replaces the document to have subsequent text edits? If not, then deleting this check should be alright.

Reply:
Logically I don't think it would make a difference anyway; either the first edit is the whole document or it's not.